lib/rbase/columns.rb

```diff
@@ -1,136 +1,120 @@
       def self.column_type(type)
         @type = type
         @@types[type] = self
       end
-      
+
       # Returns column type class that correspond to given column type string
       def self.column_for(type)
-        throw "Unknown column type '#{type}'" unless @@types.has_key?(type)
+        throw "Unknown column type '#{type}'" unless @@types.key?(type)
         @@types[type]
       end
-      
+
       # Returns column type as 1 character string
       def self.type
         @type
       end
-      
+
       # Returns column type as 1 character string
       def type
         self.class.type
       end
-      
-      # Column name
-      attr_reader :name
-      
-      # Column offset from the beginning of the record
-      attr_reader :offset
-      
-      # Column size in characters
-      attr_reader :size
-      
-      # Number of decimal places
-      attr_reader :decimal
+
+      attr_reader :name, :offset, :size, :decimal
 
       def initialize(name, options = {})
-        options.merge({:name => name, :type => self.class.type}).each { |k, v| self.instance_variable_set("@#{k}", v) }
+        options.merge(name: name, type: self.class.type)
+               .each { |k, v| instance_variable_set("@#{k}", v) }
       end
-      
-      
+
       def attach_to(table)
         @table = table
       end
 
       # Packs column value for storing it in XBase file.
-      def pack(value)
+      def pack(_value)
         throw "Not implemented"
       end
 
       # Unpacks stored in XBase column data into appropriate Ruby form.
-      def unpack(value)
+      def unpack(_value)
         throw "Not implemented"
       end
 
       def inspect
         "#{name}(type=#{type}, size=#{size})"
       end
-      
+
       protected
-      
-      def table
-        @table
-      end
-    end
 
+      attr_reader :table
+    end
 
     class CharacterColumn < Column
       column_type 'C'
-      
+
       def initialize(name, options = {})
-        if options[:size] && options[:decimal]
-          size = options[:decimal]*256 + options[:size] 
-        else
-          size = options[:size] || 254
-        end
-        
-        super name, options.merge(:size => size)
+        size = if options[:size] && options[:decimal]
+                 options[:decimal] * 256 + options[:size]
+               else
+                 options[:size] || 254
+               end
+
+        super(name, options.merge(size: size))
 
-        if options[:encoding]
+        @encoding = options[:encoding]
+        if @encoding
           @unpack_converter = Encoder.new(options[:encoding], 'utf-8')
           @pack_converter = Encoder.new('utf-8', options[:encoding])
         end
       end
-      
+
       def pack(value)
-	      value = value.to_s
-        value = @pack_converter.en(value) if @pack_converter
-        [value].pack("A#{size}")
+        @encoding ? @pack_converter.en(value) : [value.to_s].pack("A#{size}")
```

Comment:
As per unoptimised code, value in `@pack_converter.en(value)` should also be `to_s`ed before passed